protocol-units/bridge/sources/METHBridge.move

```diff
@@ -0,0 +1,455 @@
+module 0x1::METHBridge {
+    use std::signer;
+    use std::vector;
+    use aptos_framework::coin;
+    use aptos_framework::event;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::event::{EventHandle};
+    use aptos_framework::account;
+
+    const BRIDGE_ACCOUNT: address = @0x1; //Should change this later to actual bridge address
+    const EPENDING_WITHDRAWAL_NOT_FOUND: u64 = 3;
+
+    struct BridgeAccount has key {
+        deposit_events: EventHandle<DepositEvent>,
+        pending_withdrawal_events: EventHandle<PendingWithdrawalEvent>,
+        pending_withdrawals: vector<PendingWithdrawal>,
+        nonce: u256,
+    }
+
+    struct Deposit has drop, store {
+        owner: address,
+        token_id: u128,
+        nonce: u256,
+        amount: u64,
+    }
+
+    struct DepositEvent has drop, store {
+        deposit: Deposit,
+    }
+
+    struct PendingWithdrawalRequest has drop, store {
+        owner: address,
+        token_id: u128,
+        amount: u64,
+    }
+
+    struct PendingWithdrawal has drop, store {
+        request: PendingWithdrawalRequest,
+        nonce: u256,
+    }
+
+    struct PendingWithdrawalEvent has drop, store {
+        nonce: u256,
+        owner: address,
+        token_id: u128,
+        amount: u64,
+    }
+
+    public entry fun deposit(
+        trusted: &signer,
+        deposit_owner: address,
+        token_id: u128,
+        nonce: u256,
+        amount: u64
+    ) acquires BridgeAccount {
+        let trusted_address = signer::address_of(trusted);
+        assert!(trusted_address == BRIDGE_ACCOUNT, 1); // Verify trusted signer
+
+        let bridge_account = borrow_global_mut<BridgeAccount>(trusted_address);
+        let coin = coin::withdraw<AptosCoin>(trusted, amount);
```

Comment:
In the RFC its mentioned minting the asset and not depositing, unless I have got the wrong end of the stick.  What are the advantages over depositing?  One thing I am also curious about is the security of the bridge account, we are using this to secure the bridge as `trusted_address` and the funds that are bridged.  Could this maybe use another pattern?

Reply:
Yes this seemed a not-ideal to me also when implementing .. I'll have to research other bridging patterns, look at hyperlane implementations and come back with some ideas. @l-monninger let me know if you already have some other ideas.